core-bundle/src/Search/Backend/Provider/FilesProvider.php

```diff
@@ -74,22 +74,22 @@ public function updateIndex(IndexUpdateConfigInterface $trigger): iterable
         }
     }
 
-    public function convertDocumentToHit(Document $document): Hit
+    public function convertDocumentToHit(Document $document): Hit|null
     {
         // TODO: service for view and edit URLs
         $viewUrl = 'https://todo.com?view='.$document->getId();
         $editUrl = 'https://todo.com?edit='.$document->getId();
 
-        return (new Hit($document->getMetadata()['name'], $viewUrl))
+        return (new Hit($document, $document->getMetadata()['name'], $viewUrl))
```

Comment:
Is `getMetadata()['name']` always set? Otherwise

```suggestion
        return (new Hit($document, $document->getMetadata()['name'] ?? '', $viewUrl))
```

Reply:
It should be because the provider sets it itself using the row of which we know does not work without that column. But defensive is always good :)